### DISCUSSION TITLE
Cannot see contract code in forge debugger: "Contract call Unknown contract at address 0x7fa9..."

### DISCUSSION THREAD

**Discussion by TreapGoGo** [Cyfrin/foundry-full-course-cu#754] [Q&A]
In lesson 12, my test failed when I run

```bash
forge test --mt testRevertIfHealthFactorIsBrokenAfterTheMint
```

Here is my test code:

```solidity
function testRevertIfHealthFactorIsBrokenAfterTheMint() public depositCollateral {
        /**
         * 10 ETH collateral
         * 10 ETH * 2000 $/ETH = 20000$
         * can mint 20000$ * 50 / 100 = 10000$ = 10000 DSC
         *
         * If I try to mint 20000 DSC, the ending health factor is expected to be 20000 / 10000 = 2
         */
        uint256 dscToMint = 20000;
        uint256 expectedEndingHealthFactor = 2;
        vm.startPrank(USER);
        vm.expectRevert(
            abi.encodeWithSelector(DSCEngine.DSCEngine__BreaksHealthFactor.selector, expectedEndingHealthFactor)
        );
        dsce.mintDsc(dscToMint);
        vm.stopPrank();
    }

```

At first, I tried to do some static debugging, and I found that `dsce.mintDsc` didn't revert at all. But I don't know why.
So I try to debug step by step with forge debugger.

```bash
forge debug test/unit/DSCEngineTest.t.sol --sig "testRevertIfHealthFactorIsBrokenAfterTheMint()"
```

But the debugger seems to failed too. It looks like this:

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/34394212/23649054-5dc6-4bca-a4f8-111fd760bffe)

I cannot see the contract code. Instead, all I can see is those weird bytecodes. That makes me frustrated.

I would appreciate it if someone could help me out of this crashed debugger. I really enjoy debugging line by line and monitoring the variables as I go.

**Comment by Niferu** [comment]
Hi @qpdk777 

Ok the best way to debug your tests/functions etc. is just adding as many `console.log()` as you can to monitor everything. Foundry also gives you amazing tool, which is running your test like `forge test --mt testRevertIfHealthFactorIsBrokenAfterTheMint -vvvvv` this `-vvvvv` will allow you to see everything that is going on under the hood.

I will not spoil you solution, good luck ;)

**Reply by TreapGoGo** [reply]
I got it. It may help. Thanks for your answer :)

**Reply by Niferu** [reply]
@qpdk777 

Let me know if you have fixed that ;)
